src/poliastro/twobody/orbit/scalar.py

```diff
@@ -619,36 +619,48 @@ def apply_maneuver(self, maneuver, intermediate=False):
             res = orbit_new
         return res
 
-    def plot(self, label=None, use_3d=False, interactive=False):
+    def plot(self, backend_name=None, label=None):
         """Plots the orbit.
 
         Parameters
         ----------
+        backend_name : str
+            Name of the plotting backend to be used.
         label : str, optional
             Label for the orbit, defaults to empty.
-        use_3d : bool, optional
-            Produce a 3D plot, default to False.
-        interactive : bool, optional
-            Produce an interactive (rather than static) image of the orbit, default to False.
-            This option requires Plotly properly installed and configured for your environment.
 
-        """
-        if not interactive and use_3d:
-            raise ValueError(
-                "The static plotter does not support 3D, use `interactive=True`"
-            )
-        elif not interactive:
-            from poliastro.plotting.static import StaticOrbitPlotter
-
-            return StaticOrbitPlotter().plot(self, label=label)
-        elif use_3d:
-            from poliastro.plotting.interactive import OrbitPlotter3D
-
-            return OrbitPlotter3D().plot(self, label=label)
-        else:
-            from poliastro.plotting.interactive import OrbitPlotter2D
+        Returns
+        -------
+        ~poliastro.plotting.orbit.OrbitPlotter
+            An object for plotting orbits.
 
-            return OrbitPlotter2D().plot(self, label=label)
+        """
+        # HACK: avoid circular dependency
+        from poliastro.plotting.orbit.plotter import OrbitPlotter
+
+        # Select the best backend depending if it is an interactive or batch
+        # session
+        if backend_name is None:
+            try:
+                shell = get_ipython().__class__.__name__
+                if shell == "ZMQInteractiveShell":
+                    # Jupyter notebook or qtconsole
+                    is_interactive = True
+                elif shell == "TerminalInteractiveShell":
+                    # Terminal running IPython
+                    is_interactive = False
+                else:
+                    # Other type of shell
+                    is_interactive = False
+            except NameError:
+                # Standard Python interpreter
+                is_interactive = False
+            finally:
+                backend_name = "plotly2D" if is_interactive else "matplotlib2D"
```

Comment:
Oh wow, nice trick 🤯  does this work with Spyder too?

Reply:
I just checked. Both the script and the interactive Python shell in Spyder generate a "matplotlib" based figure.